finality-aleph/src/party/member.rs

```diff
@@ -10,11 +10,11 @@ use log::debug;
 use sp_runtime::traits::Block;
 
 /// Runs the member within a single session.
-pub fn task<B: Block>(
+pub fn task<B: Block, ADN: DataNetwork<AlephNetworkData<B>> + 'static>(
     subtask_common: AuthoritySubtaskCommon,
     multikeychain: KeyBox,
     config: Config,
-    network: AlephNetwork<B>,
+    network: NetworkWrapper<B, ADN>,
```

Comment:
Ugh, I don't quite remember why this cannot be `aleph_bft::Network`, could you write it out again? I remember we spent quite some time trying to make it work, but I'm not sure if the problem wasn't with some compatibility stuff, which would potentially make this doable now.

Reply:
We just agreed that having something with enormous amount of generic arguments and constraints wouldn't be that good